File: lib/models/snippet.dart
class Snippet {
  final String id;
  final String name;
  final String elementJson; // JSON of the ReadmeElement

  Snippet({
    required this.id,
    required this.name,
    required this.elementJson,
  });

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'elementJson': elementJson,
      };

  factory Snippet.fromJson(Map<String, dynamic> json) {
    return Snippet(
      id: json['id'],
      name: json['name'],
      elementJson: json['elementJson'],
    );
  }
}

File: src/models/Snippet.js
// Converted from Dart Snippet model
export class Snippet {
  constructor({ id, name, elementJson }) {
    this.id = id;
    this.name = name;
    this.elementJson = elementJson;
  }

  toJson() {
    return {
      id: this.id,
      name: this.name,
      elementJson: this.elementJson,
    };
  }

  static fromJson(json) {
    return new Snippet({
      id: json.id,
      name: json.name,
      elementJson: json.elementJson,
    });
  }
}
